play.py

```diff
@@ -96,11 +96,31 @@ def play_aidungeon_2():
 
     console_print(
         "AI Dungeon 2 will save and use your actions and game to continually improve AI Dungeon."
-        + " If you would like to disable this enter 'nosaving' for any action. This will also turn off the "
+        + " If you would like to disable this enter 'nosaving' at any time. This will also turn off the "
         + "ability to save games."
+        + "Do you want to enter 'nosaving' right now?"
     )
-
-    upload_story = True
+    choice = input("1) Yes, I want to play incognito!\n-) Press enter to skip\n>")
+    if choice == "1":
+        upload_story = False
```

Comment:
IMO, `choice` is somewhat ambiguous.

Regarding UX, I feel like this interrupts players that just want to play the game. When people open the notebook, they already have to wait to install stuff and download the model which takes a bit.

This also makes it so that user interaction is *required* before loading the model, which probably also isn't great for UX. Say somebody opens the notebook, runs the code, and does something else while they are waiting. When the user comes back, they expect it to be ready to play, and not wait for more loading.